examples/investment.jl

```diff
@@ -0,0 +1,288 @@
+using JuMP, Ipopt, GLPK
+using DualDecomposition
+using Random
+
+const DD = DualDecomposition
+
+"""
+    Scenario Tree
```

Comment:
The functions for scenario tree look written in general, which can be re-usable. @hideakiv Can you take this part and make it as part of this package by moving the part to `src`?

Reply:
I added this to ScenarioTree.jl, and included this in DualDecomposition.jl